src/utils.jl

```diff
@@ -5,12 +5,40 @@
     Assumes that `val` is globally rooted and pointer to it can be leaked. Prefer `pointer_from_objref`.
     Only use inside Enzyme.jl should be for Types.
 """
-@inline unsafe_to_pointer(val::Type{T}) where {T} = ccall(
-    Base.@cfunction(Base.identity, Ptr{Cvoid}, (Ptr{Cvoid},)),
+@inline Base.@nospecializeinfer unsafe_to_pointer(@nospecialize(val::Type)) = @static if sizeof(Int) == sizeof(Int64)
+    Base.llvmcall((
+"""
+declare nonnull {}* @julia.pointer_from_objref({} addrspace(11)*)
+
+define i64 @f({} addrspace(10)* %obj) readnone alwaysinline {
+  %c = addrspacecast {} addrspace(10)* %obj to {} addrspace(11)*
+  %r = call {}* @julia.pointer_from_objref({} addrspace(11)* %c)
```

Comment:
Huh? Doesn't this break code?

Reply:
surprisingly no (I mean this is what the implementation is supposed to do anyways) and now we don't have a cfunction ptr we have to ccall

Reply:
We should just use `Base.unsafe_pointer_to_objref`

Reply:
There was some reason historically why that wasn’t possible, eg some types saying that wasn’t valid for

Reply:
I just noticed that my brain read something correctly

```
unsafe_pointer_to_objref(x::Ptr) = ccall(:jl_value_ptr, Any, (Ptr{Cvoid},), x

function pointer_from_objref(@nospecialize(x))
    @inline
    ismutable(x) || error("pointer_from_objref cannot be used on immutable objects")
    ccall(:jl_value_ptr, Ptr{Cvoid}, (Any,), x)
en
```

So `unsafe_pointer_from_objref(x) = ccall(:jl_value_ptr, Ptr{Cvoid}, (Any,), x)`